core/src/main/java/org/apache/druid/java/util/common/JodaUtils.java

```diff
@@ -50,15 +49,25 @@
         sortedIntervals.add(interval);
       }
     }
+    return condenseIntervals(sortedIntervals.iterator());
+  }
 
-    if (sortedIntervals.isEmpty()) {
-      return new ArrayList<>();
+  /**
+   * Caller needs to insure that sortedIntervals is sorted in ascending order
```

Comment:
Can you add comment that this function does not materialize the input intervals and how it does that.

Reply:
Fixed in next commit.